localstack/services/events/target.py

```diff
@@ -0,0 +1,317 @@
+    def send_event(self, event):
+        raise NotImplementedError("Batch target is not yet implemented")
+
+    def _validate_input(self, target: Target):
+        if not collections.get_safe(target, "$.BatchParameters.JobDefinition"):
+            raise ValueError("BatchParameters.JobDefinition is required for Batch target")
+        if not collections.get_safe(target, "$.BatchParameters.JobName"):
+            raise ValueError("BatchParameters.JobName is required for Batch target")
+
+
+class ContainerTargetService(TargetService):
+    def send_event(self, event):
+        raise NotImplementedError("ECS target is not yet implemented")
+
+    def _validate_input(self, target: Target):
+        super()._validate_input(target)
+        if not collections.get_safe(target, "$.EcsParameters.TaskDefinitionArn"):
+            raise ValueError("EcsParameters.TaskDefinitionArn is required for ECS target")
+
+
+class EventsTargetService(TargetService):
+    def send_event(self, event):
+        events_client = self.client.events.request_metadata(
+            service_principal=self.service, source_arn=self.rule_arn
+        )
+        eventbus_name = self.target["Arn"].split(":")[-1].split("/")[-1]
+        source = (
+            event.get("source")
+            if event.get("source") is not None
+            else self.service
+            if self.service
+            else ""
+        )
+        detail_type = event.get("detail-type") if event.get("detail-type") is not None else ""
+        # TODO add validation and tests for eventbridge to eventbridge requires Detail, DetailType, and Source
+        # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/events/client/put_events.html
+        detail = event.get("detail", event)
+        resources = (
+            event.get("resources")
+            if event.get("resources") is not None
+            else ([self.rule_arn] if self.rule_arn else [])
+        )
+
+        events_client.put_events(
+            Entries=[
+                {
+                    "EventBusName": eventbus_name,
+                    "Source": source,
+                    "DetailType": detail_type,
+                    "Detail": json.dumps(detail),
+                    "Resources": resources,
+                }
+            ]
+        )
+
+
+class FirehoseTargetService(TargetService):
+    def send_event(self, event):
+        firehose_client = self.client.firehose.request_metadata(
+            service_principal=self.service, source_arn=self.rule_arn
+        )
+        delivery_stream_name = firehose_name(self.target["Arn"])
+        firehose_client.put_record(
+            DeliveryStreamName=delivery_stream_name, Record={"Data": to_bytes(json.dumps(event))}
+        )
+
+
+class KinesisTargetService(TargetService):
+    def send_event(self, event):
+        kinesis_client = self.client.kinesis.request_metadata(
+            service_principal=self.service, source_arn=self.rule_arn
```

Comment:
The service principal should always be events, it is the service principal making the request, not the target.

Reply:
change to initializing the service client directly to avoid requesting metadata on each send in: https://github.com/localstack/localstack/pull/10613/commits/eb3536addc6b789be2b0c88d5a098cb066c10f35

added input validation for targets requiring role arn in: https://github.com/localstack/localstack/pull/10613/commits/8cbffffd9116bed31c928efd2658b520b885a198